src/OpenSage.Game/Graphics/Cameras/Camera.cs

```diff
@@ -141,20 +141,27 @@ public Vector3 WorldToScreenPoint(Vector3 position)
             return _viewport.Project(position, Projection, View, Matrix4x4.Identity);
         }
 
-        internal Rectangle? WorldToScreenRectangle(in Vector3 position, in Size screenSize)
+        internal RectangleF? WorldToScreenRectangle(in Vector3 position, in SizeF screenSize)
         {
             var screenPosition = WorldToScreenPoint(position);
 
             // Check if point is behind camera, or too far away.
-            if (screenPosition.Z < _viewport.MinDepth || screenPosition.Z > _viewport.MaxDepth)
+            if (!IsWithinViewportDepth(screenPosition))
+            {
                 return null;
+            }
 
-            return new Rectangle(
-                (int) (screenPosition.X - screenSize.Width / 2.0f),
-                (int) (screenPosition.Y - screenSize.Height / 2.0f),
+            return new RectangleF(
+                screenPosition.X - screenSize.Width / 2.0f,
+                screenPosition.Y - screenSize.Height / 2.0f,
                 screenSize.Width, screenSize.Height);
         }
 
+        internal bool IsWithinViewportDepth(Vector3 screenPosition)
```

Comment:
This should have “in”.

Reply:
Probably not as numerics vectors are not read-only structs. One could use ref, it would look ugly, but not actually slow it down.

Reply:
I thought that because we're only accessing fields on `Vector3`, not methods, the compiler would be clever enough to not make a defensive copy. But that's only a guess.

Reply:
Yes that's true, the question is if there will be a breaking change in the future.